### ISSUE TITLE
Nullability of ref/out parameters should be treated similar to regular parameters

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#29954]
```
        // PROTOTYPE(NullableReferenceTypes): Should not report warning for
        // c.ToString(); // 3
        [Fact]
        public void UnassignedOutParameterClass()
        {
            var source =
@"class C
{
    static void G(out C? c)
    {
        c.ToString(); // 1
        c = null;
        c.ToString(); // 2
        c = new C();
        c.ToString(); // 3
    }
}";
            var comp = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition });
            comp.VerifyDiagnostics(
                // (5,9): error CS0269: Use of unassigned out parameter 'c'
                //         c.ToString(); // 1
                Diagnostic(ErrorCode.ERR_UseDefViolationOut, "c").WithArguments("c").WithLocation(5, 9),
                // (7,9): warning CS8602: Possible dereference of a null reference.
                //         c.ToString(); // 2
                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c").WithLocation(7, 9),
                // (9,9): warning CS8602: Possible dereference of a null reference.
                //         c.ToString(); // 3
                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c").WithLocation(9, 9));
        }
```

Observed:
Unexpected warning:
```
// (9,9): warning CS8602: Possible dereference of a null reference.
                //         c.ToString(); // 3
                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "c").WithLocation(9, 9));
```

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/29958

**Comment by jcouv** [comment]
I think @cston fixed this already with https://github.com/dotnet/roslyn/pull/33076
Can you confirm and close?
